src/Makefile

```diff
@@ -21,22 +21,35 @@ CL_LIB=cl-crypt
 
 CHACHA_IF_DIR:=crypt-if
 
-CL_HEADER_DIR:=opencl-platform
+# for Windows NVIDIA cygwin sample
+#CL_HEADER_DIR:=/cygdrive/c/Program\ Files/NVIDIA\ GPU\ Computing\ Toolkit/CUDA/v10.0/include/
+#CL_LIBRARY_DIR:=/cygdrive/c/Program\ Files/NVIDIA\ GPU\ Computing\ Toolkit/CUDA/v10.0/lib/x64
+# for Ubuntu Linux AMD sample
+#CL_HEADER_DIR:=/opt/amdgpu-pro/include/
+#CL_LIBRARY_DIR:=/opt/amdgpu-pro/lib/x86_64-linux-gnu/
+
+CL_PLATFORM_DIR:=opencl-platform
 
 CUDA_HEADER_DIR:=cuda-headers
 CUDA_SHA256_DIR:=cuda-sha256
 
 CUDA_DIR ?= /usr/local/cuda
 
 CXX ?= g++
+CFLAGS+=-DENDIAN_NEUTRAL -DLTC_NO_ASM -I$(CUDA_HEADER_DIR) -I$(CUDA_SHA256_DIR)
+
+# some instances require optimisations disabled (e.g. AMD on Linux)
+# remove "-cl-opt-disable" for default Optimisations to take place
+CL_DEVICE_CFLAGS=-cl-opt-disable
```

Comment:
ok; probably should be runtime detected

Reply:
In the sense of available as environment variable ?
The problem is not so much about detection, but about policy - what to apply on specific platform.